src/app/features/issue/providers/azuredevops/azuredevops-common-interfaces.service.ts

```diff
@@ -0,0 +1,163 @@
+import { Injectable } from '@angular/core';
+import { Observable, of, timer } from 'rxjs';
+import { Task } from 'src/app/features/tasks/task.model';
+import { catchError, concatMap, first, map, switchMap } from 'rxjs/operators';
+import { IssueServiceInterface } from '../../issue-service-interface';
+import { AzuredevopsApiService } from './azuredevops-api.service';
+import { ProjectService } from '../../../project/project.service';
+import { SearchResultItem } from '../../issue.model';
+import { AzuredevopsCfg } from './azuredevops.model';
+import {
+  AzuredevopsIssue,
+  AzuredevopsIssueReduced,
+} from './azuredevops-issue/azuredevops-issue.model';
+import { truncate } from '../../../../util/truncate';
+import { getTimestamp } from '../../../../util/get-timestamp';
+import { isAzuredevopsEnabled } from './is-azuredevops-enabled.util';
+import {
+  AZUREDEVOPS_INITIAL_POLL_DELAY,
+  AZUREDEVOPS_POLL_INTERVAL,
+} from './azuredevops.const';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class AzuredevopsCommonInterfacesService implements IssueServiceInterface {
+  constructor(
+    private readonly _azuredevopsApiService: AzuredevopsApiService,
+    private readonly _projectService: ProjectService,
+  ) {}
+
+  pollTimer$: Observable<number> = timer(
+    AZUREDEVOPS_INITIAL_POLL_DELAY,
+    AZUREDEVOPS_POLL_INTERVAL,
+  );
+
+  isBacklogPollingEnabledForProjectOnce$(projectId: string): Observable<boolean> {
+    return this._getCfgOnce$(projectId).pipe(
+      map((cfg) => this.isEnabled(cfg) && cfg.isAutoAddToBacklog),
+    );
+  }
+
+  isIssueRefreshEnabledForProjectOnce$(projectId: string): Observable<boolean> {
+    return this._getCfgOnce$(projectId).pipe(
+      map((cfg) => this.isEnabled(cfg) && cfg.isAutoPoll),
+    );
+  }
+
+  isEnabled(cfg: AzuredevopsCfg): boolean {
+    return isAzuredevopsEnabled(cfg);
+  }
+
+  issueLink$(issueId: number, projectId: string): Observable<string> {
+    return this._getCfgOnce$(projectId).pipe(
+      map((cfg) => `https://azuredevops.com/${cfg.organization}/issues/${issueId}`),
```

Comment:
This also needs to be configurable/use the configured base url for hosted/on-premise Azure DevOps Server instances, otherwise the link is broken... 🙃

Reply:
Also I am unsure what is different about our on-premise Azure DevOps instance (maybe the template?), but we have a different URL format and the above one does not work. It's `https://{host}/tfs/{collection|organisation}/{project}/_workitems/edit/{issueId}`.

Collections: https://learn.microsoft.com/en-us/azure/devops/server/admin/manage-project-collections?view=azure-devops-2022
Projects: https://learn.microsoft.com/en-us/azure/devops/organizations/projects/create-project?view=azure-devops-2022&tabs=browser

Reply:
This is Azure DevOps Server 2020 Update 1.1. Also tried the URL on Azure DevOps Server 2022 (AzureDevOpsServer_20221122.1) and it does not work either.

Reply:
Yeah, this link it totally wrong, will not work in hosted either.